### PR TITLE
Add ability to filter users based on groups they are in

### PR DESCRIPTION
**Pull request by wsoula**
We put admins in the AWSAdmin group so this functionality allows us to find admin users without mfa devices with the following filter
```
- type: group
   key: 'GroupName'
   value: 'AWSAdmin'
- type: credential
   key: mfa_active
   value: false
```

### CODE REVIEW

File: c7n/resources/iam.py [hunk 1]
```diff
@@ -695,6 +695,52 @@ def process(self, resources, event=None):
...
+    def user_groups(self, user_set):
+        client = local_session(self.manager.session_factory).client('iam')
+        for u in user_set:
+            if 'c7n:Groups' not in u:
+                u['c7n:Groups'] = []
+            aps = client.list_groups_for_user(
+                UserName=u['UserName'])['Groups']
+            for ap in aps:
+                u['c7n:Groups'].append(
+                    client.get_group(GroupName=ap['GroupName'])['Group'])
```
Comment (kapilt) [inline c7n/resources/iam.py]: wondering if a) we can omit the group fetch, ie the most common match here is on name which is already present b) the groups are going to be common across users, so there's a lot of redundant fetching  on the groups.

**Comment by wsoula** [comment]
@kapilt group fetch should be removed now

**Comment by wsoula** [comment]
What do I need to do about the coverage dropping from the merge of master into my branch?

**Comment by wsoula** [comment]
@kapilt is there anything else I can do to help get this PR moving forward?

**Comment by kapilt** [comment]
I think there's a minor additional optimization to avoid fetching groups for a user if we already have done so, and a unit test for the filter. I'll handle both in a followup.

**Pull request merged**